### ISSUE TITLE
Time zone Issue?

### ISSUE THREAD

**Issue by Adoncia** [HabitRPG/habitica#1457]
Since the big changeover it seems like habit is not in the right time zone. Dailies set for only Friday are showing up on Saturday morning etc. I am in GMT+8 time zone, so maybe habit isn't taking users' clock time into account or something?

**Comment by paglias** [comment]
Could you try setting dailies one day behind? I mean if you don't want a daily active on friday do not select "Thursday". In my case this is solving the issue, tell me if it works for you.

@lefnire it seems that dailies now follows this week order M, T, W, Th, F, S, Su which here in italy at least is normal but on the website they are shown following the anglo-saxon order where the week starts on sunday

**Comment by paglias** [comment]
I gave a look to the code on habitrpg-shared and I think that this bug and all the similar to this are given by the fact that week is set with Sunday as the first day but the code seem to follow the nation order (here in Italy start day is Monday) probably because habitrpg-shared is also executed on the browser and moment.js follow the browser locale. Just a supposition.

**Comment by lefnire** [comment]
this could very likely be the case. When I was running tests against cron, when I set the timezone in the tests, the tests failed until I updated which day was the beginning of the week. Since this issue is so inconsistent for everyone, I wonder indeed if it only happens on the end / beginning of the week. To be frank though, I'm a bit in over my head to try to fix this any time soon. The last time I dove into cron & timezones lead me to to a very red "Don't Drink Alcohol" habit

**Comment by paglias** [comment]
It should happen every day but me too noticed this only today, don't know
if this was because I never used this feature or due to the rewrite
Il giorno 08/set/2013 00:30, "Tyler Renelle" notifications@github.com ha
scritto:

**Comment by lefnire** [comment]
good to know. well, nothing in cron has changed in the rewrite that wasn't updated right when we launched mobile. So this will have been effecting people _at least_ since mobile release. I know people have been reporting issues to @MagicMickey , so it was probably introduced the moment we added timezones to the mix (aka, cron calculated on the server instead of the client). I'll put this on the "prouction release" milestone and get to it probably last of all. I just hate these timezone / cron tickets so bad, I need to de-stress on some other tickets for a while
